### DISCUSSION TITLE
うちのRailsのdockerのサイズが1.5GBあって困ってます。

### DISCUSSION THREAD

**Discussion by scrpgil** [roppongirb/discussions#33] [Ideas]

**Comment by yusukesugasawa** [comment]
Roppongi.rb#16 用コメント

**Comment by hatappi** [comment]
## 概要
Rails を docker 使ってコンテナイメージにして Kubernetes 上で Run!
リリースする時にイメージが大きくて？交換が遅い 😢 

そもそもどれくらいになるのが適切？？？

**Comment by hatappi** [comment]
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

**Reply by scrpgil** [reply]
RailsのDockerfileのサンプル。
https://docs.docker.com/samples/rails/

**Comment by hatappi** [comment]
## Ruby 特有でやってること
- 本番で使う gem のみに絞るようにする
  - rspec とか
- bundle install, assets:precompile はどこでやる？
  - bundle install はどちらにしても必要なので Dockerfile に構築しても良さそう
  - asseets:sync で S3 にアップロードしちゃう！
    - image 単体で動かないという懸念もあり 😢

**Reply by hatappi** [reply]
asset_sync
https://github.com/AssetSync/asset_sync

**Comment by hatappi** [comment]
## Ruby 以外な話
- キャッシュだったりが含まれてたりする場合もあるので、まずは layer ごとの差分を確認するか実際のコンテナに入って大きい部分を見つけるなど **計測する！**

**Reply by scrpgil** [reply]
```
COPY . .
```
